src/node.h

```diff
@@ -197,16 +197,10 @@ typedef intptr_t ssize_t;
 
 namespace node {
 
-NODE_EXTERN extern bool no_deprecation;
-#if HAVE_OPENSSL
-NODE_EXTERN extern bool ssl_openssl_cert_store;
-# if NODE_FIPS_MODE
-NODE_EXTERN extern bool enable_fips_crypto;
-NODE_EXTERN extern bool force_fips_crypto;
-# endif
-#endif
+// Forward References
+class NodeOption;
 
-NODE_EXTERN int Start(int argc, char *argv[]);
+NODE_EXTERN int Start(int argc, char *argv[], NodeOption* state);
```

Comment:
How is an embedder going to call this without access to the class?

Reply:
This is still to be finished... I'll likely export this via node.h but obviously open to suggestions :-)

Reply:
@jasnell The current design of `NodeOption` would make every change an ABI breakage if it’s public API, we probably don’t want this.

I’d see something like this as the goal:

```c++
std::shared_ptr<NodeOptions> ParseOptions(int argc, const char* const* argv);
```

And then being able to pass that pointer to Node’s methods that need it. How does that sound?

Reply:
> E.g. should there be a programmatic way of building the object?

I don’t know for sure, but the fact that V8 doesn’t need it seems to be a sign that this is “programmatic” enough.

It might be nice to be able to modify an existing `NodeOptions` object by parsing more arguments, but I don’t think that’s something we *need* to worry about right now?

Reply:
Ok. I was going to move the ParseArgs in a separate PR but can do it here instead.